source/SIL.AppBuilder.Portal.Frontend/src/ui/components/header/notifications/display.tsx

```diff
@@ -1,91 +1,92 @@
+
+import { NotificationResource } from '~/data';
 
-  clearAll = (e) => {
-    const { clearAll } = this.props;
+import { useCollectionDataActions } from '~/data/containers/resources/notification/with-collection-data-actions';
+
+import { preventDefault } from '~/lib/dom';
+
+import { useLiveData } from '~/data/live';
+
+interface ISubscribedTo {
+  notifications: NotificationResource[];
+}
 
-    e.preventDefault();
+export default function Notifications({ refetch }) {
+  const { t } = useTranslations();
+  const [visible, toggleVisible] = useToggle(false);
+  const {
+    subscriptions: { notifications },
+  } = useOrbit<ISubscribedTo>({
+    notifications: (q) => q.findRecords('notification').sort('-dateCreated', '-dateRead'),
+  });
 
-    clearAll();
+  const { clearAll, markAllAsViewed } = useCollectionDataActions(notifications);
+
+  const haveAllNotificationsBeenSeen = notifications.every(
+    (n) => attributesFor(n).dateRead !== null
+  );
+  const hasNotifications = notifications.length > 0;
+
+  const toggle = () => {
+    if (visible) {
+      markAllAsViewed();
+      refetch();
+    }
+
+    toggleVisible();
   };
 
-  render() {
-    const { t } = this.props;
-
-    const {
-      notifications,
-      haveAllNotificationsBeenSeen,
-      isThereAtLeastOneNotificationToShow,
-    } = this.props;
-
-    const hasNotifications =
-      notifications && notifications.length > 0 && isThereAtLeastOneNotificationToShow;
-
-    const isMenuVisible = this.state.visible ? 'visible' : '';
-
-    return (
-      <div className='ui top right pointing dropdown' data-test-header-notification>
-        <div
-          style={{ zIndex: 1 }}
-          className={`full-overlay transition ${this.state.visible ? 'visible invisible' : ''}`}
-          onClick={this.toggle}
-        />
-
-        <div
-          data-test-notification-trigger
-          data-test-notification-active={!haveAllNotificationsBeenSeen}
-          style={{ position: 'relative' }}
-          onClick={this.toggle}
-        >
-          {haveAllNotificationsBeenSeen ? <NotificationIcon /> : <NotificationActiveIcon />}
-        </div>
-
-        <div className={`ui menu transition notifications ${isMenuVisible}`}>
-          {hasNotifications && (
-            <>
-              <div className='notification-buttons'>
-                <a href='#' data-test-clear-all onClick={this.clearAll}>
-                  {t('header.clearAll')}
-                </a>
-              </div>
-
-              <div className={notifications.length > 3 ? 'scrollable-menu' : ''}>
-                {notifications.map((notification) => (
-                  <Row key={notification.id} notification={notification} />
-                ))}
-              </div>
-            </>
-          )}
-
-          {!hasNotifications && (
-            <div className='notification-no-data'>{t('header.emptyNotifications')}</div>
-          )}
-        </div>
+  const isMenuVisible = visible ? 'visible' : '';
```

Comment:
a variable name like `isMenuVisible` makes me think boolean but instead, this is defining the class to apply

Reply:
totally agree.
It was just pulled from here: https://github.com/sillsdev/appbuilder-portal/pull/567/files#diff-26e4c95cb718005691cad1c85e16de6bL46